pkg/cmd/psa/psa.go

```diff
@@ -0,0 +1,291 @@
+package psa
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	psapi "k8s.io/pod-security-admission/api"
+
+	"github.com/spf13/cobra"
+)
+
+// PSAOptions contains all the options and configsi for running the PSA command.
+type PSAOptions struct {
+	level      string
+	namespaces []string
+
+	client   *kubernetes.Clientset
+	warnings *warningsHandler
+}
+
+var (
+	psaExample = `
+	# Check if your namespaces could be upgraded to the restricted level.
```

Comment:
exit code should indicate if we passed or not.

Reply:
I am not sure that I understood it without more context. Would you like me to exit with a non-zero code if we have violations?